### PR TITLE
pythonPackages.wavedrom: init at 2.0.3.post2

### PR DESCRIPTION
**Pull request by airwoodix**
###### Motivation for this change

Add [wavedrom](https://pypi.org/project/wavedrom/) (https://github.com/wallento/wavedrompy) for Python.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137313` at fb25bbdd run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.wavedrom</li>
    <li>python39Packages.wavedrom</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/development/python-modules/wavedrom/default.nix:53:16:
```
   |
53 |     homepage = https://github.com/wallento/wavedrompy;
   |                ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-explicit-check-phase.md">python-explicit-check-phase</a>

Consider using `pytestCheckHook` in `checkInputs` rather than `checkPhase = "pytest";`.

Near pkgs/development/python-modules/wavedrom/default.nix:45:3:
```
   |
45 |   checkPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/wavedrom/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  checkInputs = [
+    pytestCheckHook
+    xmldiff
+    pillow
+    cairosvg
+  ];
+
+  postPatch = ''
+    sed -i 's|use_scm_version=True|version="${version}"|' setup.py
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/wavedrom/default.nix:41]: Please use SETUPTOOLS_SCM_PRETEND_VERSION env instead.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/wavedrom/default.nix (hunk 1 above)
Comment (airwoodix) [reply pkgs/development/python-modules/wavedrom/default.nix]: Thanks for the hint! Fixed in b0555f0.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 137313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.wavedrom</li>
<li>python39Packages.wavedrom</li>
</ul>
</details>

**Pull request merged**